main.go

```diff
@@ -271,12 +278,28 @@ func executeArg(cmd *cobra.Command, arg string, w io.Writer) error {
 }
 
 func executeCLI(cmd *cobra.Command, src *source, w io.Writer) error {
-	b, err := io.ReadAll(src.reader)
-	if err != nil {
-		return fmt.Errorf("unable to read from reader: %w", err)
-	}
-
-	b = utils.RemoveFrontmatter(b)
+	// Create context with signal handling for graceful shutdown
```

Comment:
Most/all signal handling dance needs to be moved to the very top `main` function.

Reply:
fixed